src/client/mod.rs

```diff
@@ -1031,13 +1032,43 @@ impl IotHubClient {
         //   - failed
         //   - timed out
         self.confirmation_set.spawn(async move {
-            match timeout(Duration::from_secs(Self::CONFIRMATION_TIMEOUT_SECS), rx).await {
+            match timeout(Duration::from_secs(Self::get_confirmation_timeout()), rx).await {
                 Ok(Ok(false)) => panic!("twin_report: failed"),
                 Err(_) => panic!("twin_report: timed out"),
                 _ => trace!("twin_report: succeeded"),
             }
         });
     }
+
+    fn get_confirmation_timeout() -> u64 {
+        static INIT: Once = Once::new();
+        static mut CONFIRMATION_TIMEOUT_IN_SECS: u64 = CONFIRMATION_TIMEOUT_DEFAULT_IN_SECS;
+
+        unsafe {
```

Comment:
why is that in an unsafe closure?

Reply:
> why is that in an unsafe closure?

Thus I avoid using a mutex. Rust doesn't allow static mutable vars without synchronization. Since I write only once at this line it is save...  (code is inspired by official docu)